File: api/supertest-typescript/tests/api.test.ts
import request from 'supertest';

const baseUrl = process.env.BASE_URL || 'http://localhost:3000';
const api = request(baseUrl);

describe('API Smoke Tests', () => {
  it('smoke: GET /api/health returns 200', async () => {
    const res = await api.get('/api/health');
    expect(res.status).toBe(200);
  });

  it('smoke: GET /api/users returns array', async () => {
    const res = await api.get('/api/users').expect(200);
    expect(Array.isArray(res.body)).toBe(true);
  });

  it('POST /api/users creates a user', async () => {
    const res = await api
      .post('/api/users')
      .send({ name: 'Test User', email: 'test@example.com' })
      .expect(201);
    expect(res.body).toHaveProperty('id');
  });

  it('GET /api/users/:id returns user', async () => {
    const res = await api.get('/api/users/1').expect(200);
    expect(res.body).toHaveProperty('name');
  });

  it('returns 404 for unknown route', async () => {
    await api.get('/api/nonexistent').expect(404);
  });
});

File: api/rest-assured-java/src/test/java/tests/ApiTest.java
package tests;

import config.Config;
import io.restassured.RestAssured;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;

import static io.restassured.RestAssured.*;
import static org.hamcrest.Matchers.*;

@Tag("smoke")
public class ApiTest {

    @BeforeAll
    public static void setup() {
        RestAssured.baseURI = Config.getBaseUrl();
    }

    @Test
    void health_check_returns_200() {
        get("/api/health").then().statusCode(200);
    }

    @Test
    void get_users_returns_array() {
        get("/api/users").then()
                .statusCode(200)
                .body("$", instanceOf(java.util.List.class));
    }

    @Test
    void create_user() {
        given()
                .contentType("application/json")
                .body("{\"name\":\"Test User\",\"email\":\"test@example.com\"}")
        .when()
                .post("/api/users")
        .then()
                .statusCode(201)
                .body("id", notNullValue());
    }

    @Test
    void unknown_route_returns_404() {
        get("/api/nonexistent").then().statusCode(404);
    }
}
